smt.py

```diff
@@ -324,7 +352,11 @@ def make_smtlib(p: assume_prove.AssumeProveProg) -> SMTLIB:
         source.ExprVar(source.type_bool, p.entry))))
 
     cmds.append(CmdCheckSat())
-    return merge_smtlib(emit_cmd(cmd) for cmd in cmds)
+    emitted: list[SMTLIB] = []
+    emitted.append(emit_prelude())
```

Comment:
do you wanna do `cmds.extend(emit_prelude())`, where emit_prelude emits a bunch of `Cmd`s rather than emitting some text? Like this we can keep structured data for as long as possible (this will be required in the future because we will look at the emitted commands to know what response we expect in return).